src/main.cpp

```diff
@@ -2930,6 +2930,22 @@ static DisconnectResult DisconnectBlock(const CBlock& block, CValidationState& s
         }
     }
 
+    // Grab the latest subtree (according to the view) and use it
+    // to determine if the block being disconnected was responsible
+    // for completing a subtree. If so, we'll pop the subtree.
```

Comment:
```suggestion
    // for completing a subtree. If so, we'll pop the subtree.
    // (It is not possible for a block to complete more than one subtree, due to
    // the maximum number of outputs/actions in a block being less than 2^16.)
```

Reply:
[Tangent; not necessary for this PR, but maybe nice-to-have in the future.] This is the kind of "constraint-at-a-distance" that makes me nervous.

One day suppose we want to transition to giant blocks with >> 2^16 actions. To do that transition successfully, we'd need to know every place in either design or code impl that implicitly relies on actions < 2^16.

How would we search for those "constraint dependencies"? This comment helps if we're doing textual searches and guess the right phrases.

It seems like a more rigorous solution would be an explicit compile-time assertion akin to `MAX_BLOCK_ACTIONS < COMPLETE_SUBTREE_LEAF_COUNT`; during the hypothetical giant-block development, when the constraint were relaxed we'd immediately see a compilation error. Can C++ do that?

Reply:
We could use a `static_assert` to establish that invariant, yes.